airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java

```diff
@@ -16,21 +22,28 @@ public interface SqlGenerator<DialectTableDefinition> {
    * Generate a SQL statement to create a fresh table to match the given stream.
    * <p>
    * The generated SQL may throw an exception if the table already exists. Callers should use
-   * {@link #alterTable(StreamConfig, java.lang.Object)} if the table is known to exist.
+   * {@link #existingSchemaMatchesStreamConfig(StreamConfig, java.lang.Object)} if the table is known to exist.
    *
    * @param suffix A suffix to add to the stream name. Useful for full refresh overwrite syncs, where
    *        we write the entire sync to a temp table.
    */
   String createTable(final StreamConfig stream, final String suffix);
 
   /**
-   * Generate a SQL statement to alter an existing table to match the given stream.
-   * <p>
-   * The operations may differ based on the existing table definition (BigQuery does not allow
-   * altering a partitioning scheme and requires you to recreate+rename the table; snowflake only
-   * allows altering some column types to certain other types, etc.).
+   * Check the final table's schema and compare it to what the stream config would generate.
+   * @param stream the stream/stable in question
+   * @param existingTable the existing table mapped to the stream
+   * @return whether the existing table matches the expected schema
+   * @throws TableNotMigratedException if the table does not contain all {@link SqlGenerator#FINAL_TABLE_AIRBYTE_COLUMNS}
+   */
+  boolean existingSchemaMatchesStreamConfig(final StreamConfig stream, final DialectTableDefinition existingTable) throws TableNotMigratedException;
+
+  /**
+   * SQL Statements which will rebuild the final table using the raw table data
+   * @param stream the stream to rebuild
+   * @return an ordered sequence of SQL statements to execute to rebuild the final table.
    */
-  String alterTable(final StreamConfig stream, DialectTableDefinition existingTable);
+  List<String> softReset(final StreamConfig stream);
```

Comment:
FWIW I think we probably _should_ make this optimization but I'm going to hold off for now since the naive implementation of concating with newlines and `;` is failing 😢